src/corehost/cli/hostpolicy/hostpolicy.cpp

```diff
@@ -100,19 +131,26 @@ int get_or_create_coreclr(
             return StatusCode::Success;
```

Comment:
Can you please get rid of this two stage lock here?
We first try without grabbing the lock and then we grab it and try again.
I understand that it's an optimization, but in this case, the perf hit of grabbing a mutex which should have basically no contention on it is negligible and it will reduce the code complexity (especially the fact that the TODO belongs in two places... should be just one).

Reply:
Naturally goes away with putting `coreclr_t` on `hostpolicy_context_t`